File: SDKs/googleapis.com/shoppingcontent/v2/go/pkg/models/shared/testordercustomer.go
package shared

type TestOrderCustomer struct {
	Email                       *string                               `json:"email"`
	ExplicitMarketingPreference *bool                                 `json:"explicitMarketingPreference"`
	FullName                    *string                               `json:"fullName"`
	MarketingRightsInfo         *TestOrderCustomerMarketingRightsInfo `json:"marketingRightsInfo"`
}

File: SDKs/googleapis.com/shoppingcontent/v2/typescript/src/sdk/models/shared/testordercustomer.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import { TestOrderCustomerMarketingRightsInfo } from "./testordercustomermarketingrightsinfo";


export class TestOrderCustomer extends SpeakeasyBase {
  @Metadata({ data: "json, name=email" })
  email?: string;

  @Metadata({ data: "json, name=explicitMarketingPreference" })
  explicitMarketingPreference?: boolean;

  @Metadata({ data: "json, name=fullName" })
  fullName?: string;

  @Metadata({ data: "json, name=marketingRightsInfo" })
  marketingRightsInfo?: TestOrderCustomerMarketingRightsInfo;
}
